x-pack/plugin/core/src/test/java/org/elasticsearch/protocol/xpack/watcher/GetWatchResponseTests.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.protocol.xpack.watcher;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.protocol.AbstractHlrcStreamableXContentTestCase;
+import org.elasticsearch.xpack.core.watcher.actions.ActionStatus;
+import org.elasticsearch.xpack.core.watcher.execution.ExecutionState;
+import org.elasticsearch.xpack.core.watcher.support.xcontent.XContentSource;
+import org.elasticsearch.xpack.core.watcher.transport.actions.get.GetWatchResponse;
+import org.elasticsearch.xpack.core.watcher.watch.WatchStatus;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class GetWatchResponseTests extends
+    AbstractHlrcStreamableXContentTestCase<GetWatchResponse, org.elasticsearch.client.watcher.GetWatchResponse> {
+
+    private static final String[] SHUFFLE_FIELDS_EXCEPTION = new String[] { "watch" };
+
+    @Override
+    protected String[] getShuffleFieldsExceptions() {
+        return SHUFFLE_FIELDS_EXCEPTION;
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
+    }
+
+    @Override
+    protected List<XContentType> supportedContentTypes() {
+        // The get watch API supports JSON output only.
+        return Arrays.asList(XContentType.JSON);
+    }
+
+    @Override
+    protected GetWatchResponse createBlankInstance() {
+        return new GetWatchResponse();
+    }
+
+    @Override
+    protected GetWatchResponse createTestInstance() {
+        String id = randomAlphaOfLength(10);
+        if (rarely()) {
+            return new GetWatchResponse(id);
+        }
+        long version = randomLongBetween(0, 10);
+        WatchStatus status = randomWatchStatus();
+        BytesReference source = emptyWatch();
+        return new GetWatchResponse(id, version, status, new XContentSource(source, XContentType.JSON));
+    }
+
+    private static BytesReference emptyWatch() {
```

Comment:
It could be good to return a simple, non-empty structure here so that we test that aspect of the response parsing.